### PR TITLE
Task expiration email notification

### LINKED ISSUE #31: task expiration email notification

### PR DESCRIPTION
**Pull request by hizmailovich**
Scheduler is created, close #31

### CODE REVIEW

File: docker.env [hunk 1]
```diff
@@ -4,6 +4,8 @@ POSTGRES_HOST=postgres
 JDBC_LOG_LEVEL=DEBUG
 WEB_LOG_LEVEL=DEBUG
 APP_LOG_LEVEL=DEBUG
+MAIL_USERNAME=abialiauski.alerter@gmail.com
+MAIL_PASSWORD=jypsojwddawmkcfq
```
Comment (h1alexbel) [inline docker.env:8]: @hizmailovich can you encrypt it or store more safely?

File: src/main/java/com/taskviewer/api/scheduled/ExpirationEmailScheduler.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        List<Task> list = new ArrayList<>(); //todo this.taskService.iterate()
+        list.forEach(task -> {
+            long full = Duration.between(
+                            task.time().tracked(), task.time().estimate())
+                    .getSeconds();
+            long left = Duration.between(
+                            LocalDateTime.now(), task.time().estimate())
+                    .getSeconds();
+            if (left <= 0.2 * full) {
+                this.mailService.send(this.userService.byId(task.user()), "Expiration email",
```
Comment (h1alexbel) [inline src/main/java/com/taskviewer/api/scheduled/ExpirationEmailScheduler.java:36]: line is too long split it

File: src/main/java/com/taskviewer/api/service/MailService.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.taskviewer.api.service;
+
+import com.taskviewer.api.model.User;
+
+public interface MailService {
+
+    void send(final User user, final String subject, final String message);
```
Comment (h1alexbel) [inline src/main/java/com/taskviewer/api/service/MailService.java:7]: `final` in interfaces is not required

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: docker.env (hunk 1 above)
Comment (hizmailovich) [reply docker.env]: JavaMailSender doesn't support encrypted password to set, but we can use base64 decoder for username and password, pushed

File: src/main/java/com/taskviewer/api/service/MailServiceImpl.java [hunk 4]
```diff
@@ -25,7 +25,7 @@ public class MailServiceImpl implements MailService {
 
     @Override
     @SneakyThrows
-    public void send(final User user, final String subject, final String message) {
+    public void send(User user, String subject, String message) {
```
Comment (h1alexbel) [inline src/main/java/com/taskviewer/api/service/MailServiceImpl.java:28]: Class args should be final

**Review by h1alexbel — CHANGES_REQUESTED** [review]

**Review by h1alexbel — APPROVED** [review]

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@hizmailovich thanks!

**Pull request merged**